### PR TITLE
Add funding statement in TEI output 

### LINKED ISSUE #895: Paragraph content missing for 'Funding' (annex) section
I am using GROBID to convert bioRxiv preprints PDFs to XML and finding that paragraph content with the heading 'Funding' is not being captured in the TEI XML output (version 0.7.0, lightweight Docker image).

Example PDF:
[2021.09.27.461862v1.full.pdf](https://github.com/kermitt2/grobid/files/8154868/2021.09.27.461862v1.full.pdf)

The 'Funding' section in this document appears immediately after the 'Acknowledgements' on Page 19 of the PDF. It is captured in the TEI XML as `@type="annex"`, but the `<p>` element and text content is missing:

```
<div type="annex">
  <div 
    xmlns="http://www.tei-c.org/ns/1.0">
    <head>Funding</head>
   </div>
</div>
```
Although recognition of this section with some `@type="funding"` attribute would be the ideal scenario, for my use case I simply need the associated `<p>` content (i.e. "This work was funded by...") to be output in the TEI XML.

Many thanks in advance for looking into this issue.

### PR DESCRIPTION
**Pull request by lfoppiano**
This PR adds the funding statement (already in training data and models) in the TEI output. 

The funding output will be placed in a standardised position in the `<back>` of the TEI output, as the availability statement. 

e.g. (with sentence segmentation)

```xml
<div type="funding">
<div xmlns="http://www.tei-c.org/ns/1.0"><head>Funding</head><p><s>This study was supported by the International Collaborative Research Grants Scheme with joint grants from the Wellcome Trust UK (GR071587MA) and the Australian NHMRC (268055).</s><s>The funding sources played no role in study design, data collection, analysis or interpretation, writing the report, or the decision to submit the paper for publication.</s></p></div>
</div>
```

This should fix #895 #652 

As discussed in #957, this does not cover the case when only header is processed.

### CODE REVIEW

**Comment by kermitt2** [comment]
Thanks ! 
I think the change in [TEIFormatter.java](https://github.com/kermitt2/grobid/pull/959/files#diff-e88bc69454f8143f785f6053c65c9933fec104835298419e9f290a1ea689f44d) is doing nothing? Just moving stuff that do the same (simply put a default space at inline position of a figure to avoid agglutination of text).

**Comment by kermitt2** [comment]
From the discussion https://github.com/kermitt2/grobid/issues/956, I am updating the PR with post-processing limited to processShort(), instead  of changing the general fulltext model decoding.

**Comment by kermitt2** [comment]
The changes seem to work fine, on the example from #956 we obtain:

```xml
           <div type="funding">
                <div
                    xmlns="http://www.tei-c.org/ns/1.0">
                    <p>This study was supported by the South Asian Clinical Toxicology Research Collaboration, which is funded by The Wellcome Trust/ National Health and Medical Research Council International Collaborative Research Grant GR071669MA. The funding bodies had no role in analyzing or interpreting the data or writing the article.</p>
                </div>
            </div>
```
No runtime error on PMC_sample set and biorxiv_test_2000.

**Comment by lfoppiano** [comment]
I found several cases where where the papers has the funding statement in the back of the document, the TEI serialisation calls `getSectionAsTEI()` which does not uses the `processShort()`.

E.g. this document: [bj4360053.pdf](https://github.com/kermitt2/grobid/files/9797087/bj4360053.pdf)

Unfortunately, the result does not help, and the hack in processShort() does not work because of the different label at the first place won't fix the sequence. 

```
FUNDING	funding	F	FU	FUN	FUND	G	NG	ING	DING	BLOCKSTART	LINESTART	ALIGNEDLEFT	NEWFONT	HIGHERFONT	1	0	ALLCAP	NODIGIT	0	NOPUNCT	0	10	0	NUMBER	0	0	I-<section>
This	this	T	Th	Thi	This	s	is	his	This	BLOCKSTART	LINESTART	ALIGNEDLEFT	NEWFONT	LOWERFONT	0	0	INITCAP	NODIGIT	0	NOPUNCT	0	10	0	NUMBER	0	0	I-<table>
work	work	w	wo	wor	work	k	rk	ork	work	BLOCKIN	LINEIN	ALIGNEDLEFT	SAMEFONT	SAMEFONTSIZE	0	0	NOCAPS	NODIGIT	0	NOPUNCT	0	10	0	NUMBER	0	0	<table>
was	was	w	wa	was	was	s	as	was	was	BLOCKIN	LINEIN	ALIGNEDLEFT	SAMEFONT	SAMEFONTSIZE	0	0	NOCAPS	NODIGIT	0	NOPUNCT	0	10	0	NUMBER	0	0	<table>
supported	supported	s	su	sup	supp	d	ed	ted	rted	BLOCKIN	LINEIN	ALIGNEDLEFT	SAMEFONT	SAMEFONTSIZE	0	0	NOCAPS	NODIGIT	0	NOPUNCT	0	10	0	NUMBER	0	0	<table>
[...]
```

I'm not sure how I could fix this properly

**Comment by kermitt2** [comment]
We could wait for future versions which will remove the tables and figures from the fulltext model, or apply the same fix as in processShort for every table and figure labels (not just the ones starting the sequence). What do you think?

**Comment by kermitt2** [comment]
So the two problematic cases are working fine with the current version, yeah :D 

I cleaned the previous hack in `processShort()` and replaced it with `postProcessFullTextLabeledText()`, to be sure it is always applied after calling `processShort()` (so this applied to the funding statement in header too now, otherwise it was only applied to the funding statement at the end of a paper).

**Comment by kermitt2** [comment]
Thanks !!
